third_party/zlib/CMakeLists.txt

```diff
@@ -22,8 +22,8 @@ set(
 ExternalProject_Add(zlib-
   PREFIX ${CMAKE_CURRENT_BINARY_DIR}
   DOWNLOAD_DIR ${ARCHIVE_DIR}
-  URL https://zlib.net/zlib-1.2.11.tar.gz
-  URL_HASH SHA256=c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1
+  URL https://zlib.net/zlib-1.2.12.tar.gz
+  URL_HASH SHA256=91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9
```

Comment:
Although this is correct. Can we actually keep 1.2.11 and instead update the URL to one on sourceforge? i.e. `URL https://sourceforge.net/projects/libpng/files/zlib/1.2.11/zlib-1.2.11.tar.gz`

Here are the reasons.
1. We learned that zlib.net would delete previous releases when there is a new release. This gives concerns because when there is a new release of zlib, then the source build would break again. Due to this, the latest versioned release (0.11) cannot be built from source. If we rely on sourceforge.net, then we can expect that old releases are preserved. 
2. Currently there is no 1.2.12 on sourceforge. https://sourceforge.net/projects/libpng/files/zlib/ We need to wait for them to upload 1.2.12 to source forge.